src/fides/api/models/privacy_preference.py

```diff
@@ -102,36 +99,204 @@ class PrivacyPreferenceHistory(Base):
             padding="pkcs5",
         ),
     )
-    # Whether the user wants to opt in, opt out, or has acknowledged the notice
-    preference = Column(EnumColumn(UserConsentPreference), nullable=False, index=True)
+
     # The specific version of the experience config the user was shown to present the relevant notice
     # Contains the version, language, button labels, description, etc.
-    privacy_experience_config_history_id = Column(
-        String,
-        ForeignKey("privacyexperienceconfighistory.id"),
-        nullable=True,
-        index=True,
-    )
+    @declared_attr
+    def privacy_experience_config_history_id(cls) -> Column:
+        return Column(
+            String,
+            ForeignKey("privacyexperienceconfighistory.id"),
+            nullable=True,
+            index=True,
+        )
+
     # The specific experience under which the user was presented the relevant notice
     # Minimal information stored here, mostly just region and component type
-    privacy_experience_id = Column(
-        String, ForeignKey("privacyexperience.id"), nullable=True, index=True
-    )
+    @declared_attr
+    def privacy_experience_id(cls) -> Column:
+        return Column(
+            String, ForeignKey("privacyexperience.id"), nullable=True, index=True
+        )
+
     # The specific historical record the user consented to
-    privacy_notice_history_id = Column(
-        String, ForeignKey(PrivacyNoticeHistory.id), nullable=False, index=True
+    @declared_attr
+    def privacy_notice_history_id(cls) -> Column:
+        return Column(
+            String, ForeignKey(PrivacyNoticeHistory.id), nullable=False, index=True
+        )
+
+    # Optional FK to a verified provided identity (like email or phone), if applicable
+    @declared_attr
+    def provided_identity_id(cls) -> Column:
+        return Column(String, ForeignKey(ProvidedIdentity.id), index=True)
+
+    # Location where we received the request
+    request_origin = Column(EnumColumn(RequestOrigin))  # privacy center, overlay, API
+
+    url_recorded = Column(String)
+    user_agent = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+    )
+
+    user_geography = Column(String, index=True)
+
+    # Relationships
+    @declared_attr
+    def privacy_notice_history(cls) -> relationship:
+        return relationship(PrivacyNoticeHistory)
+
+    @declared_attr
+    def provided_identity(cls) -> relationship:
+        return relationship(ProvidedIdentity, foreign_keys=[cls.provided_identity_id])
+
+    @declared_attr
+    def fides_user_device_provided_identity(cls) -> relationship:
+        return relationship(
+            ProvidedIdentity, foreign_keys=[cls.fides_user_device_provided_identity_id]
+        )
```

Comment:
FYI I am defining FK fields and relationship fields on a Mixin class (which doesn't have a table created for it) - defining with this declared_attr property allows me to do this.